utils/post.js

```diff
@@ -368,8 +375,17 @@ async function filterFeeds(
   feeds = [],
   feed_id = null,
   threshold = null,
-  isBlurredPost = false
+  isBlurredPost = false,
+  followingUsers = []
 ) {
+  // get all actor karma score
+  let actors = [];
+  feeds.map((item) => {
+    if (!actors.includes(item.actor.id)) {
+      actors.push(item.actor.id);
+    }
+  });
+  const karmaScores = await UsersFunction.getUsersKarmaScore(User, actors);
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In this section, you are using a map function to populate the `actors` array. However, since you're not transforming the data and don't need a new array, it would be more appropriate to use a forEach loop instead of map. Also, consider using a Set to automatically handle duplicate actor ids.
```diff
- let actors = [];
- feeds.map((item) => {
-   if (!actors.includes(item.actor.id)) {
-     actors.push(item.actor.id);
-   }
- });
+ let actors = new Set();
+ feeds.forEach((item) => {
+   actors.add(item.actor.id);
+ });
+ actors = Array.from(actors);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->